capstone/capdb/models.py

```diff
@@ -979,17 +987,23 @@ def get_frontend_url(self, cite=None, disambiguate=False, include_host=True):
             If disambiguate is true, return /series/volnum/pagenum/id/.
         """
         cite = cite or self.citations.first()
-        # try to match "(volnumber) (series) (pagenumber)"
-        m = re.match(r'(\S+)\s+(.+?)\s+(\S+)$', cite.cite)
+
+        # try to match "(volnumber) (Series) (pagenumber)"
+        m = re.match(r'^(\S+)\s+([A-Z].+?)\s+(\S+)$', cite.cite)
```

Comment:
Is it worth adding special handling for the `81 1/2` and `33 Suppl.` cases here? The regex could look like:

`m = re.match(r'^(\S+(?: Suppl\.| 1/2)?)\s+([A-Z].+?)\s+(\S+)$', cite.cite)`

This way we don't end up setting `disambiguate = True` for those two volumes unnecessarily.

Reply:
Done